op-challenger/game/fault/trace/super/provider.go

```diff
@@ -0,0 +1,134 @@
+	"github.com/ethereum/go-ethereum/log"
+)
+
+var (
+	ErrGetStepData = errors.New("GetStepData not supported")
+	ErrIndexTooBig = errors.New("trace index is greater than max uint64")
+
+	InvalidTransition     = []byte("invalid")
+	InvalidTransitionHash = crypto.Keccak256Hash(InvalidTransition)
+)
+
+const (
+	StepsPerTimestamp = 1024
+)
+
+type RootProvider interface {
+	// TODO: Should we redefine SuperRootResponse so we don't depend on supervisor?
+	SuperRootAtTimestamp(timestamp uint64) (supervisortypes.SuperRootResponse, error)
+}
+
+type SuperTraceProvider struct {
+	types.PrestateProvider
+	logger             log.Logger
+	rootProvider       RootProvider
+	prestateTimestamp  uint64
+	poststateTimestamp uint64
+	l1Head             eth.BlockID
+	gameDepth          types.Depth
+}
+
+func NewSuperTraceProvider(logger log.Logger, prestateProvider types.PrestateProvider, rootProvider RootProvider, l1Head eth.BlockID, gameDepth types.Depth, prestateTimestamp, poststateTimestamp uint64) *SuperTraceProvider {
+	return &SuperTraceProvider{
+		PrestateProvider:   prestateProvider,
+		logger:             logger,
+		rootProvider:       rootProvider,
+		prestateTimestamp:  prestateTimestamp,
+		poststateTimestamp: poststateTimestamp,
+		l1Head:             l1Head,
+		gameDepth:          gameDepth,
+	}
+}
+
+func (s *SuperTraceProvider) Get(ctx context.Context, pos types.Position) (common.Hash, error) {
+	// Find the timestamp and step at position
+	timestamp, step, err := s.ComputeStep(pos)
+	if err != nil {
+		return common.Hash{}, err
+	}
+	s.logger.Info("Getting claim", "pos", pos.ToGIndex(), "timestamp", timestamp, "step", step)
+	if step == 0 {
+		root, err := s.rootProvider.SuperRootAtTimestamp(timestamp)
+		if err != nil {
+			return common.Hash{}, fmt.Errorf("failed to retrieve super root at timestamp %v: %w", timestamp, err)
+		}
+		return common.Hash(root.SuperRoot), nil
+	}
+	// Fetch the super root at the next timestamp since we are part way through the transition to it
+	prevRoot, err := s.rootProvider.SuperRootAtTimestamp(timestamp)
+	if err != nil {
+		return common.Hash{}, fmt.Errorf("failed to retrieve super root at timestamp %v: %w", timestamp, err)
+	}
+	nextTimestamp := timestamp + 1
+	nextRoot, err := s.rootProvider.SuperRootAtTimestamp(nextTimestamp)
+	if err != nil {
+		return common.Hash{}, fmt.Errorf("failed to retrieve super root at timestamp %v: %w", nextTimestamp, err)
+	}
+	prevChainOutputs := make([]eth.ChainIDAndOutput, 0, len(prevRoot.Chains))
+	for _, chain := range prevRoot.Chains {
+		prevChainOutputs = append(prevChainOutputs, eth.ChainIDAndOutput{ChainID: chain.ChainID.ToBig().Uint64(), Output: chain.Canonical})
+	}
+	expectedState := interopTypes.TransitionState{
+		SuperRoot:       eth.NewSuperV1(prevRoot.Timestamp, prevChainOutputs...).Marshal(),
+		PendingProgress: make([]interopTypes.OptimisticBlock, 0, step),
+		Step:            step,
+	}
+	for i := uint64(0); i < min(step, uint64(len(prevChainOutputs))); i++ {
```

Comment:
Shouldn't we include padded steps once each chain outputs are filled in?

Reply:
For padded steps we update the step but don't append anything else to the pending progress array (because there's no block to add).  We set the step correctly when creating the transition state on line 87 so only need to copy over the pending blocks here.

Reply:
got it. Thanks for explaining.